src/internal/recovery/recovery.go

```diff
@@ -0,0 +1,50 @@
+// Package recovery implements subsystem-independent disaster recovery.
+package recovery
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+	"github.com/pachyderm/pachyderm/v2/src/version"
+)
+
+type SnapshotID int64
+
+// String implements fmt.Stringer.
+func (id SnapshotID) String() string {
```

Comment:
For my understanding: Will this String() be ever passed in a id<1? Consider that is invalid and should have raised error in upper stack.